src/simple_cache_client.rs

```diff
@@ -163,66 +163,55 @@ impl SimpleCacheClientBuilder {
-    // ```
-    pub fn new(auth_token: String, default_ttl: Duration) -> MomentoResult<Self> {
-        SimpleCacheClientBuilder::new_with_explicit_agent_name(auth_token, default_ttl, "sdk", None)
-    }
-
-    /// Like new() above, but requires a momento_endpoint.
-    // TODO: Update the documentation and tests and deprecate the existing new method. This will be
-    // done once we start vending out tokens with no endpoints and have published the new momento
-    // endpoints.
-    pub fn new_with_endpoint(
-        auth_token: String,
+    /// ```
+    /// # tokio_test::block_on(async {
+    ///     use momento::{CredentialProviderBuilder, SimpleCacheClientBuilder};
+    ///     use std::env;
+    ///     use std::time::Duration;
+    ///     let credential_provider = CredentialProviderBuilder::new_from_environment_variable("TEST_AUTH_TOKEN")
+    ///         .build()
+    ///         .expect("TEST_AUTH_TOKEN must be set");
+    ///     let momento = SimpleCacheClientBuilder::new(credential_provider, Duration::from_secs(30))
+    ///         .expect("could not create a client")
+    ///         .build();
+    /// # })
+    /// ```
+    pub fn new(
+        credential_provider: CredentialProvider,
         default_ttl: Duration,
-        momento_endpoint: String,
     ) -> MomentoResult<Self> {
         SimpleCacheClientBuilder::new_with_explicit_agent_name(
-            auth_token,
+            credential_provider,
             default_ttl,
             "sdk",
-            Some(momento_endpoint),
         )
     }
 
     /// Like new() above, but used for naming integrations.
     pub fn new_with_explicit_agent_name(
-        auth_token: String,
+        credential_provider: CredentialProvider,
         default_ttl: Duration,
         user_agent_name: &str,
-        momento_endpoint: Option<String>,
     ) -> MomentoResult<Self> {
-        let momento_endpoints =
-            match MomentoEndpointsResolver::resolve(&auth_token, momento_endpoint) {
-                Ok(endpoints) => endpoints,
-                Err(e) => return Err(e),
-            };
-        log::debug!("connecting to endpoints: {:?}", momento_endpoints);
+        log::debug!(
+            "connecting to cache endpoint: {:?}, connecting to control endpoint: {:?}",
+            &credential_provider.cache_endpoint,
+            &credential_provider.control_endpoint
+        );
 
-        let control_channel = connect_channel_lazily(&momento_endpoints.control_endpoint.url)?;
-        let data_channel = connect_channel_lazily(&momento_endpoints.data_endpoint.url)?;
+        let control_channel = connect_channel_lazily(&credential_provider.control_endpoint)?;
+        let data_channel = connect_channel_lazily(&credential_provider.cache_endpoint)?;
 
         match utils::is_ttl_valid(default_ttl) {
             Ok(_) => Ok(Self {
-                data_endpoint: momento_endpoints.data_endpoint.hostname,
+                data_endpoint: credential_provider.cache_endpoint,
```

Comment:
The only place this gets used is by the create and signing key APIs - the response includes the endpoint the key is for, and that's pulled from this.

Previously, this would look like (from CLI output):
```
$ momento -p alpha preview signing-key list
[
  {
    "key_id": "123456",
    "expires_at": {
      "secs_since_epoch": 1691206339,
      "nanos_since_epoch": 0
    },
    "endpoint": "cache.cell.momentohq.com"
  }
]
```

with this change, it will look instead like


```
[
  {
    "key_id": "123456",
    "expires_at": {
      "secs_since_epoch": 1691206339,
      "nanos_since_epoch": 0
    },
    "endpoint": "https://cache.cell.momentohq.com"
  }
]
```

I don't think this is an issue, but I'm not completely sure about that?

Reply:
It looks like the scheme is the only difference here? I think that should be okay.